src/lib/types.h

```diff
@@ -106,6 +106,12 @@ typedef struct {
     const char *string;
 } pgp_map_t;
 
+/** multi-precision integer, used in signatures and public/secret keys */
+typedef struct pgp_mpi_t {
```

Comment:
Why are we introducing a new type for this? Shouldn't we be using `bignum_t`, which is backed by secure memory in botan?

Reply:
The idea is that all data is in 'raw' format once it is parsed from the source/written out. And conversion to bignum_t is done only during cryptography operation. Say, we don't need convert to bignum_t all 20 keys we read from the keyring, only one which is used for verification or signing.

Reply:
Ok that sounds reasonable. I think once we start using C++ in here we ought to use `Botan::secure_vector` for this, just to make sure things are wiped on free (I see it's manually done here) and mlock'd.